src/main/java/seedu/planner/model/ModulePlanner.java

```diff
@@ -447,6 +435,140 @@ private void sortAvailableModules(List<Module> modulesAvailable, UserProfile use
         return allModules;
     }
 
+    private Optional<ModuleDescription> getModuleDescription(String code) {
+        return MajorDescription.getModuleCode(userProfile.getMajor(), code);
+    }
+
+    /**
+     * Count the number of modules fulfilling degree requirement which is not
+     * a University Level Requirement or Breadth and Depth.
+     *
+     * @param degreeRequirement the degree requirement
+     * @return the number of modules fulfilling that degree requirement
+     */
+    private int countProgrammeRequirementNonBd(DegreeRequirement degreeRequirement) {
```

Comment:
I'm a little unsure on my suggestion. I think we can just name this as `countProgrammeRequirement` and let the comment set the preconditions of the degreeRequirements not being ULR or BD. Should be safe to do so since it's a private method and is not likely to be misused. I first thought of adding `NonUlr` but that would too long haha